pkg/kn/commands/service/import.go

```diff
@@ -0,0 +1,150 @@
+
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"knative.dev/client/pkg/kn/commands"
+	clientservingv1 "knative.dev/client/pkg/serving/v1"
+
+	clientv1alpha1 "knative.dev/client/pkg/apis/client/v1alpha1"
+)
+
+// NewServiceImportCommand returns a new command for importing a service.
+func NewServiceImportCommand(p *commands.KnParams) *cobra.Command {
+	command := &cobra.Command{
+		Use:   "import FILENAME",
+		Short: "Import a service and its revisions",
+		Example: `
+ # Import a service in YAML format
+ kn service import /path/to/file.yaml
+
+ # Import a service in JSON format
+ kn service import /path/to/file.json`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errors.New("'kn service import' requires filename of import file as single argument")
+			}
+			filename := args[0]
+
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+
+			client, err := p.NewServingClient(namespace)
+			if err != nil {
+				return err
+			}
+
+			return importWithOwnerRef(client, filename, cmd.OutOrStdout())
+		},
+	}
+	flags := command.Flags()
+	commands.AddNamespaceFlags(flags, false)
+
+	return command
+}
+
+func importWithOwnerRef(client clientservingv1.KnServingClient, filename string, out io.Writer) error {
+	var export clientv1alpha1.Export
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	decoder := yaml.NewYAMLOrJSONDecoder(file, 512)
+	err = decoder.Decode(&export)
+	if err != nil {
+		return err
+	}
+	serviceName := export.Spec.Service.Name
+
+	// Return error if service already exists
+	svcExists, err := serviceExists(client, serviceName)
+	if err != nil {
+		return err
+	}
+	if svcExists {
+		return fmt.Errorf("cannot import service '%s' in namespace '%s' because the service already exists",
+			serviceName, client.Namespace())
+	}
+
+	fmt.Fprintf(out, "Importing Service '%s' in namespace '%s'.\n", serviceName, client.Namespace())
+
+	// Create service with empty TrafficTarget to avoid missing revision errors
+	tmpSvc := export.Spec.Service
+	tmpSvc.Spec.Traffic = []servingv1.TrafficTarget{}
+	if err = client.CreateService(&tmpSvc); err != nil {
+		return err
+	}
+
+	// Retrieve current Configuration to be use in OwnerReference
+	currentConf, err := client.GetConfiguration(serviceName)
+	if err != nil {
+		return err
+	}
+
+	// Create revision with current Configuration's OwnerReference
+	if len(export.Spec.Revisions) > 0 {
+		for _, r := range export.Spec.Revisions {
+			tmp := r
+			// OwnerRef ensures that Revisions are recognized by controller
+			tmp.OwnerReferences = []metav1.OwnerReference{
+				{
+					APIVersion:         currentConf.APIVersion,
+					Kind:               currentConf.Kind,
+					Name:               currentConf.Name,
+					UID:                currentConf.UID,
+					Controller:         ptr.Bool(true),
+					BlockOwnerDeletion: ptr.Bool(true),
```

Comment:
pls check whether you can use the `kmeta` package to set the owner reference. Below is from the serving repo
https://github.com/knative/serving/blob/5233f0c607c5a89d6a58d6faac9eed57b4877b79/pkg/reconciler/revision/resources/deploy.go#L253

Reply:
Nice find, thanks! It's exactly the function needed here.